### PR TITLE
Breaking Change - Removed ApiVersion parameter from Az.Resources deployments cmdlets and changed the output type of Get-AzResourceGroupDeploymentOperations

### PR DESCRIPTION
**Pull request by Xynoclafe**
## Description
Removed ApiVersion parameter from Az.Resources deployments cmdlets and changed the output type of Get-AzResourceGroupDeploymentOperations

This is a breaking change.
Design review - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/707

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by filizt** [comment]
>     public abstract class ResourceManagerCmdletBaseWithAPiVersion : ResourceManagerCmdletBase

Could you please fix the casing of "APi" in the class name?

---
Refers to: src/Resources/ResourceManager/Implementation/CmdletBase/ResourceManagerCmdletBaseWithAPiVersion.cs:21 in 6c9887e. [](commit_id = 6c9887e516eebacce92f68329da7586b61e24ed8, deletion_comment = False)

File: src/Resources/Resources/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Updated `Get-AzResourceGroupDeploymentOperation`to use the SDK.
+* Remove ApiVersion parameter
```
Comment (filizt) [inline src/Resources/Resources/ChangeLog.md:22]: I think it's a worthy effort to make the change longs a bit verbose to make things clear. We can add something like "Remove ApiVersion parameter from Deployment cmdlets (if the cmdlet list is not too long, maybe add them too?)

**Comment by filizt** [comment]
Could you also add the link to the design discussion in the description section?

File: src/Resources/Resources/ChangeLog.md (hunk 1 above)
Comment (Xynoclafe) [reply src/Resources/Resources/ChangeLog.md:22]: There are quite a few cmdlets (mentioned in the design review), so I don't think adding them to the changelog is practical. But I'll make it a little more verbose.

**Comment by Xynoclafe** [comment]
@isra-fel Can you please review this PR?

**Review by isra-fel — APPROVED** [review]
Sorry @Xynoclafe our team are just back from holiday. Anyway your changes LGTM. Thanks

**Comment by Xynoclafe** [comment]
Hi @isra-fel
Can you review the PR again? Your last commit voided your previous review. And I think we can remove the _Waiting for CI_ label as CI seems to be passing now.

**Comment by isra-fel** [comment]
@Xynoclafe I'm sorry but it seems the merge of #13090 has caused a lot of conflict in the test recording file. Would you please resolve them by merging master and record `ResourceGroupDeploymentEndToEnd` again?

Thanks in advance!

**Review by isra-fel — APPROVED** [review]

**Pull request merged**